### PR TITLE
mgr/dashboard: Change icon color in notifications

### PR DESCRIPTION
**Pull request by votdev**
- Use CSS filters to convert the existing icons to white.
- Add new Prometheus SVG image, the previous PNG image has used the color white instead or transparent to fill some areas.

Before:
53229843-9c291880-3685-11e9-9452-cdfa3ea76c46

After:
bildschirmfoto von 2019-02-22 11-33-46

bildschirmfoto von 2019-02-25 15-41-32
(Example how the Prometheus icon looks like, ignore the message text)

Signed-off-by: Volker Theile <vtheile@suse.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/styles.scss [hunk 1]
```diff
@@ -344,6 +344,13 @@ uib-accordion .panel-title,
 .ceph-icon {
   background: url('assets/Ceph_Logo_Stacked_RGB_120411_fa_228x228.png');
 }
+.toast-message .ceph-icon {
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/src/styles.scss]: This doesn't work for all icons in use.

**Comment by LenzGr** [comment]
The ceph dashboard tests failed, but `mgr-dashboard-frontend-unittests` actually passed. So I think this is good to go.

File: src/pybind/mgr/dashboard/frontend/src/styles.scss (hunk 1 above)
Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/styles.scss]: Currently there are only 2 icons as far as i can see. The Prometheus icon is already white, so only the Ceph icon must be modified.

Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/styles.scss]: My initial idea was NOT to display all icons using white color (maybe other upcoming icons would better match with green and the orther notification background colors). But maybe it is better to display ALL icons using white for better visibility.

Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/styles.scss]: Fixed

Comment (Devp00l) [reply src/pybind/mgr/dashboard/frontend/src/styles.scss]: Hm... you are right, a white dot isn't any better (happens with your current change). So please revert the change or use a different prometheus icon with no white infill. Like https://github.com/prometheus/docs/blob/master/static/prometheus_logo_orange_circle.svg or https://github.com/prometheus/docs/blob/master/static/prometheus_logo_orange_invert.svg

File: src/pybind/mgr/dashboard/frontend/src/styles.scss [hunk 2]
```diff
@@ -365,13 +365,23 @@ h3.page-header {
...
 }
+
+/* Notifications */
+.toast-message .custom-icon {
+  /* Display icons in white, regardless of their origin color. */
+  filter: brightness(0) invert(1);
+  -webkit-filter: brightness(0) invert(1);
+  -moz-filter: brightness(0) invert(1);
+  -o-filter: brightness(0) invert(1);
+  -ms-filter: brightness(0) invert(1);
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/styles.scss]: It seems that without prefix `filter` [covers 90% of the users](https://caniuse.com/#feat=css-filters) and if you check the usual desktop browsers it's almost 100%. For the 0-10% remaining, the effect would just be a red-coloured logo instead, so not a big deal. I was wondering if the common practice (web-dev rookie here) is to generate new UI assets via CSS filters or process and save them as new files.

If we want to generate cross-browser CSS, why not using some library? Just googled and found these:
- https://github.com/Schepp/CSS-Filters-Polyfill
- https://github.com/pascalduez/SassyFilters
- https://github.com/iamvdo/pleeease-filters

File: src/pybind/mgr/dashboard/frontend/src/styles.scss [hunk 3]
```diff
@@ -365,13 +365,23 @@ h3.page-header {
...
   background-color: white;
   border: 1px solid grey;
   color: #333;
   font-size: 1.1em;
 }
+
+/* Notifications */
+.toast-message .custom-icon {
+  /* Display icons in white, regardless of their origin color. */
+  filter: brightness(0) invert(1);
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/styles.scss]: Nevertheless, I really liked the trick!

File: src/pybind/mgr/dashboard/frontend/src/styles.scss (hunk 2 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/styles.scss]: Angular CLI already does that: https://angular.io/guide/build#configuring-browser-compatibility.
I try to remove all the `filter` styles with prefixes and the CLI automatically added the `webkit-filter`.

We could remove the filters here, but I'm guessing we have lots of prefixes throughout the rest of the code.
A separate PR would be my recommendation, and we could also change the settings to get more coverage.

Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/styles.scss]: I think using a library to handle the filters is too much for something really trivial. IMHO, if there are prefixes that are not used anymore or only for a small number of supported browser, so what.

File: src/pybind/mgr/dashboard/frontend/src/styles.scss (hunk 1 above)
Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/styles.scss]: I've replaced the Prometheus PNG logo with the SVG counterpart which renders the transparent areas correctly.

File: src/pybind/mgr/dashboard/frontend/src/styles.scss (hunk 2 above)
Comment (epuertat) [reply src/pybind/mgr/dashboard/frontend/src/styles.scss]: Thanks, @votdev.

**Pull request merged**